### ISSUE TITLE
Add a third option which replaces newlines by spaces when pasting content with multiple lines

### ISSUE THREAD

**Issue by FremyCompany** [microsoft/terminal#9400]
# Description of the new feature/enhancement

When pasting in the Terminal a command line that ends with a line break, my options are either to paste and run the command, or abort the operation. Most of the times, I actually want to paste while trimming the trailing line break, such that I can edit the command I am pasting.

![image](https://user-images.githubusercontent.com/364405/110218078-ede11c00-7eb7-11eb-8527-1ca33ea88f01.png)

# Proposed technical implementation details (optional)

When pasting, a new button would appear in the confirmation dialog, that would allow to paste while trimming line breaks (if the string contains line breaks that are not trimmed, replace them by a space?)

**Comment by FremyCompany** [comment]
Adding a third button might require some refactoring, because the dialog is only meant to have two buttons:

https://github.com/microsoft/terminal/blob/f26c246c7c081c0222dd8ebd4738f6a0d574f66c/src/cascadia/TerminalApp/TerminalPage.cpp#L2145-L2165

**Comment by skyline75489** [comment]
I don't hate this idea. But with bracketed paste feature in 1.7, this does not really bother me when I'm using bash/zsh. The shell will not execute the command and let you edit it. Sadly powershell does not support bracketed paste, though.

**Comment by zadjii-msft** [comment]
I don't think this is a terrible idea. Looks like the `ContentDialog` supports 3 buttons, it would just take a bit of refactoring.
![image](https://user-images.githubusercontent.com/18356694/110317076-3e986800-7fd1-11eb-84c0-82ffa683a5d9.png)

**Comment by Nacimota** [comment]
I've just taken a crack at this and I think I've got it working without any substantial refactoring at all. I'm still pretty intimidated by C++ though, so I'm not sure if I'm doing anything stupid/careless, but I'm happy to submit this in a PR if anyone wants to take a look.

**Comment by chenzhiwei** [comment]
Could you simply add an option to terminal, the values can be: 1. trim trailing newline, 2. don't trim, 3, interactive dialog

**Comment by zadjii-msft** [comment]
Merging some discussion from #10502

> I do think we should be more intelligent here... text that is a single line and ends in a newline with no further content should just strip it by default, I think. I am opposed to adding a button to every paste dialog that folks will have to process when they paste. Perhaps there's a middle ground!

**Comment by Nacimota** [comment]
> Who's really wanting to paste all these lines with spaces between them (instead of newlines) here?

After thinking about it myself for a bit, I am inclined to agree with you. My PR on this is pretty stale anyway because we hadn't decided what the button text should be and then I think I just forgot about it. But yes, I am struggling to think of a situation where you would actually want to strip breaks from *the middle* of the paste. I suppose maybe if you copied from some poorly formatted source (PDFs, for example, move in mysterious ways) wherein the command was supposed to be one line but for some reason contains line breaks? Feels like a stretch, though.

Anecdotally, this warning only shows up for me because either I am indeed pasting multiple commands or I just accidentally copied the wrong thing.

> `trimSingleTrailingNewline` (default:true) for "if there's just one newline at the end then trim that", so folks could set this to false to be able to paste a command in one go?

Automatically trimming the trailing new line makes sense to me, but I wonder does it really matter if it's just one? As long as there's no other characters in between, multiple line breaks at the end of the string seems practically equivalent to a single one to me.

**Comment by FremyCompany** [comment]
Single trailing new line stripping sounds good to me as well.

**Comment by zadjii-msft** [comment]
Talked with the team today - we do feel like the right way forward is just to trim the trailing newline if that's the only one in the paste contents. To quote: "It doesn't make sense to be warned for multiple lines, but copy a single line and try to paste it AND ONLY SEE ONE LINE IN THE DIALOG. That's dumb".

**Comment by shtratos** [comment]
Huge thanks @lovef, @zadjii-msft and all the contributors! 
I was pulling my hair out every time I was trying to paste a command into a terminal. Can't wait for the next release!
